addons/mail/static/src/core/common/store_service.js

```diff
@@ -93,19 +73,116 @@ export class Store extends BaseStore {
     odoobotOnboarding;
     users = {};
     internalUserGroupId = null;
-    registeredImStatusPartners = null;
+    imStatusTrackedPersonas = Record.many("Persona", {
+        compute() {
+            return Object.values(this.Persona?.records ?? []).filter(
+                (persona) =>
+                    persona.type === "partner" &&
+                    persona.im_status !== "im_partner" &&
+                    !persona.is_public
+            );
+        },
+        onUpdate() {
+            this.updateImStatusRegistration();
+        },
+        eager: true,
+    });
     hasLinkPreviewFeature = true;
     // messaging menu
     menu = { counter: 0 };
+    menuThreads = Record.many("Thread", {
+        /** @this {import("models").Store} */
+        compute() {
+            /** @type {import("models").Thread[]} */
+            let threads = Object.values(this.Thread.records).filter(
+                (thread) =>
+                    thread.displayToSelf ||
+                    (thread.needactionMessages.length > 0 && thread.type !== "mailbox")
+            );
+            const tab = this.discuss.activeTab;
+            if (tab !== "main") {
+                threads = threads.filter(({ type }) => this.tabToThreadType(tab).includes(type));
+            } else if (tab === "main" && this.env.inDiscussApp) {
+                threads = threads.filter(({ type }) =>
+                    this.tabToThreadType("mailbox").includes(type)
+                );
+            }
+            return threads;
```

Comment:
We might need to split this further, so that we don't have to recompute all the threads again when changing tab.

We also might want to compute as many things as possible on the One (inverse) so thread in this case, so that we don't have to use a list in a compute.